src/ngx_http_lua_socket_tcp.c

```diff
@@ -4891,6 +4967,7 @@ ngx_http_lua_socket_tcp_upstream_destroy(lua_State *L)
         return 0;
     }
 
+    luaL_unref(L, LUA_REGISTRYINDEX, u->bind_tbl_ref);
```

Comment:
Also need a test case for the idle connections with attached data in pool get closed when still in the pool.